mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java

```diff
@@ -0,0 +1,224 @@
+
+        testingData = MessageIdManagerWithPreDeletionHooksTestSystemProvider
+            .createTestingData(sessionProvider, mapperFactory, ImmutableSet.of(messageVaultHook));
+
+        mailboxManager = testingData.getMailboxManager();
+        messageIdManager = testingData.getMessageIdManager();
+
+        internalDate = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss").parse("2019/02/26 14:30:00");
+        mailContent = Message.Builder.of()
+            .setSubject("test")
+            .setBody("testmail", StandardCharsets.UTF_8)
+            .setSender(ALICE_ADDRESS)
+            .setTo(TEST_ADDRESS)
+            .setDate(internalDate)
+            .build();
+
+        searchQuery = new SearchQuery();
+        searchQuery.andCriteria(SearchQuery.internalDateOn(internalDate, SearchQuery.DateResolution.Second));
+    }
+
+    @Test
+    void notifyDeleteShouldAppendMessageVault() throws Exception {
+        Mailbox mailbox = testingData.createMailbox(MAILBOX_ALICE_ONE, aliceSession);
+        MessageManager messageManager = testingData.createMessageManager(mailbox, aliceSession);
+        MessageId messageId = appendMessage(messageManager).getMessageId();
+
+        messageIdManager.delete(ImmutableList.of(messageId), aliceSession);
+
+        DeletedMessage deletedMessage = buildDeletedMessage(ImmutableList.of(mailbox.getMailboxId()), messageId, ALICE);
+        assertThat(messageVault.search(ALICE, Query.ALL).collectList().block().get(0))
+            .isEqualTo(deletedMessage);
+    }
+
+    @Test
+    void notifyDeleteShouldAppendMessageToOwnerVaultWhenSharedUserDeleteMessageInSharingMailbox() throws Exception {
+        Mailbox mailboxOne = testingData.createMailbox(MAILBOX_ALICE_ONE, aliceSession);
+        MessageManager aliceMessageManager = testingData.createMessageManager(mailboxOne, aliceSession);
+        MessageManager bobMessageManager = testingData.createMessageManager(mailboxOne, bobSession);
+        ComposedMessageId composedMessageId = appendMessage(aliceMessageManager);
+        MessageId messageId = composedMessageId.getMessageId();
+
+        mailboxManager.setRights(MAILBOX_ALICE_ONE,
+            MailboxACL.EMPTY.apply(MailboxACL.command()
+                .forUser(BOB_ADDRESS)
+                .rights(MailboxACL.Right.Lookup, MailboxACL.Right.Read, MailboxACL.Right.DeleteMessages, MailboxACL.Right.PerformExpunge)
+                .asAddition())
+            ,aliceSession);
+
+        DeletedMessage deletedMessage = buildDeletedMessage(ImmutableList.of(mailboxOne.getMailboxId()), messageId, ALICE);
+        bobMessageManager.delete(ImmutableList.copyOf(bobMessageManager.search(searchQuery, bobSession)), bobSession);
+
+        assertThat(messageVault.search(ALICE, Query.ALL).collectList().block().get(0))
+            .isEqualTo(deletedMessage);
+    }
+
+    @Test
+    void notifyDeleteShouldAppendMessageToSharedUserVaultWhenSharedUserDeleteMessageAfterMoveToAnotherMailbox() throws Exception {
```

Comment:
What is a shared user vault ?

Reply:
It is `Vault` of shared user of sharing mailbox. I change it to `notifyDeleteShouldNotAppendMessageToVaultOfSharedUserOfMailboxWhenSharedUserDeleteMessageInSharingMailbox()`

Reply:
Sorry but I still don't understand. What is a shared user ? I only know shared mailboxes.

Reply:
Alice share a mailbox for Bob -> I call Bob as `Shared user`.
But now, I change to OtherUser